### PR TITLE
Answers to Search (4): Rename config variables

### PR DESCRIPTION
**Pull request by alextaing**
Rename the following:
AnswersConfig -> SearchConfig
BaseAnswersConfig -> BaseSearchConfig
AnswersConfigWithApiKey -> SearchConfigWithApiKey
AnswersConfigWithToken -> SearchConfigWithToken
AnswersConfigWithDefaulting -> SearchConfigWithDefaulting

### CODE REVIEW

File: src/models/deprecated.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { SearchConfig } from './core/SearchConfig';
+import { BaseSearchConfig } from './core/SearchConfig';
+import { SearchConfigWithApiKey } from './core/SearchConfig';
+import { SearchConfigWithToken } from './core/SearchConfig';
+
+/**
+ * @deprecated AnswersConfig is deprecated and has been replaced by SearchConfig
```
Comment (oshi97) [inline src/models/deprecated.ts:7]: Could you try using an `{@link SearchConfig}` so the documentation pages have a link? also below

File: src/models/core/SearchConfig.ts [hunk 2]
```diff
@@ -40,10 +40,10 @@ export interface BaseAnswersConfig {
 
 /**
  * Configuration options for {@link AnswersCore}, which includes the
```
Comment (nmanu1) [inline src/models/core/SearchConfig.ts:42]: are these links to `AnswersCore` updated in one of the other PRs?

File: src/models/core/SearchConfig.ts [hunk 3]
```diff
@@ -71,16 +71,16 @@ export interface AnswersConfigWithToken extends BaseAnswersConfig {
...
+ * For a full description of the options, see {@link BaseSearchConfig}.
  * The config requires either an apiKey or a token.
  *
  * @public
  */
-export type AnswersConfig = AnswersConfigWithApiKey | AnswersConfigWithToken;
+export type SearchConfig = SearchConfigWithApiKey | SearchConfigWithToken;
 
 /**
  * The AnswersConfig after any defaulting has been done from within answers-core.
```
Comment (nmanu1) [inline src/models/core/SearchConfig.ts:82]: will there be another PR for updating docs like these?

File: src/models/core/SearchConfig.ts (hunk 2 above)
Comment (alextaing) [reply src/models/core/SearchConfig.ts:42]: They will be added in another PR.  Since I renamed SearchCore in an old PR and not in this one, it would point to nothing and cause an error :(

**Review by nmanu1 — APPROVED** [review]
